ERR_PNPM_NO_OFFLINE_TARBALL for various PNPM packages

This issue is for solving and discussing the recent PNPM build failures with the `ERR_PNPM_NO_OFFLINE_TARBALL` message. It includes packages like `typespec`, `renovate`, `zipline`

Some prior info can be found in these PRs:
- https://github.com/NixOS/nixpkgs/pull/425610
- https://github.com/NixOS/nixpkgs/pull/426407

There is a similar [bug](https://github.com/pnpm/pnpm/issues/5315) that is related to `pnpm deploy`, this issue (should) have nothing to do with that

--- a/pkgs/development/tools/pnpm/fetch-deps/pnpm-config-hook.sh
+++ b/pkgs/development/tools/pnpm/fetch-deps/pnpm-config-hook.sh
@@ -36,6 +36,10 @@ pnpmConfigHook() {
 
     pnpm config set store-dir "$STORE_PATH"
 
+    # Prevent hard linking on file systems without clone support.
+    # See: https://pnpm.io/settings#packageimportmethod
+    pnpm config set package-import-method clone-or-copy
+
     if [[ -n "$pnpmWorkspace" ]]; then
         echo "'pnpmWorkspace' is deprecated, please migrate to 'pnpmWorkspaces'."
         exit 2

